map3d.py

```diff
@@ -0,0 +1,133 @@
+        phi_2 = topo_limits[3]
+        theta_1 = topo_limits[0]
+        theta_2 = topo_limits[1]
+    
+    #find indices of topo limits
+    phi_deg = data['y']
+    theta_deg = data['x']
+    phi_ind1 = np.argmin(np.abs(phi_deg-phi_1))
+    phi_ind2 = np.argmin(np.abs(phi_deg-phi_2))
+    theta_ind1 = np.argmin(np.abs(theta_deg-theta_1))
+    theta_ind2 = np.argmin(np.abs(theta_deg-theta_2))
+
+    #restrict topo extent
+    phi=phi[phi_ind1:phi_ind2]
+    theta=theta[theta_ind1:theta_ind2]
+    c = c[theta_ind1:theta_ind2:,phi_ind1:phi_ind2]
+    phi, theta = np.meshgrid(phi,theta)
+   
+
+    #prep data grid
+    phi_iso, theta_iso = np.meshgrid(((ydata*np.pi*2)/360.)+np.pi/2.,(xdata*np.pi*2)/360.)
+ 
+    if mode is 'sphere':
+        x = np.sin(phi) * np.cos(theta[::-1]) * (1 + c/zscale)
+        y = np.sin(phi) * np.sin(theta[::-1]) * (1 + c/zscale)
+        z = np.cos(phi) * (1 + c/30000.)
+    
+        x_iso = np.sin(phi_iso) * np.cos(theta_iso[::-1]) * (1 -zdata/zscale)
+        y_iso = np.sin(phi_iso) * np.sin(theta_iso[::-1]) * (1 -zdata/zscale)
+        z_iso = np.cos(phi_iso) * (1 -zdata/zscale)
+    
+    elif mode is 'cylinder':
+        x = np.sin(phi) * np.cos(theta[::-1])
+        y = np.sin(phi) * np.sin(theta[::-1])
+        z = c/zscale
+
+        x_iso = np.sin(phi_iso) * np.cos(theta_iso[::-1])
+        y_iso = np.sin(phi_iso) * np.sin(theta_iso[::-1])
+        z_iso = -zdata/zscale
+    
+    elif mode is 'rectangle':
+        y, x = np.meshgrid(phi_deg[phi_ind1:phi_ind2],theta_deg[theta_ind1:theta_ind2])
+        z = c/zscale
+    
+        y_iso,z_iso = np.meshgrid(ydata,zdata)
+        x_iso,z_iso = np.meshgrid(xdata,zdata) 
+        z_iso =-z_iso/zscale
+
+    if topo_vmin is None:
+        tvmin = -6000
+    else:
+        tvmin = topo_vmin
+    if topo_vmax is None:
+        tvmax = 0
+    else:
+        tvmax = topo_vmax
+    
+    #make figure
+    mlab.figure(size = (1024,768),bgcolor = (1,1,1), fgcolor = (0.5, 0.5, 0.5))
+    mlab.clf()
+    # Plot Bathymetry mesh
+    m = mlab.mesh(x, y, z, colormap=topo_cmap,vmin=tvmin/zscale,vmax=tvmax/zscale)
+    
+    #optional: reverse bathymetry colormap
+    if topo_cmap_reverse is True:
+        lut = m.module_manager.scalar_lut_manager.lut.table.to_array() 
+        ilut=lut[::-1]
+        m.module_manager.scalar_lut_manager.lut.table = ilut
+
+    #optional: plot constant color on land
+    if land_constant is True:
+        sl = mlab.mesh(x, y, z,mask = z<0,color =land_color)
+
+
+    #plot data surface
+    m = mlab.mesh(x_iso, y_iso, z_iso,scalars=scalardata,colormap=data_cmap,vmin =vmin,vmax=vmax,opacity=data_alpha)
+    m.module_manager.scalar_lut_manager.lut.nan_color = [0,0,0,0]
+    
+    #optional: change mayavi camera settings
+    if set_view is None:
+        mlab.view(distance = 'auto')
+    else:
+        mlab.view(azimuth = set_view[0], elevation = set_view[1], distance = set_view[2], focalpoint = set_view[3])
+    mlab.show()
+
+    return
```

Comment:
rather than returning `None`, why don't you return `mlab`? That way you can can separate out the `show()` part, and your function can be more composable and testable.